### ISSUE TITLE
[Bug Report]: Version requirements don't allow major.minor.patch

### ISSUE THREAD

**Issue by flanakin** [Azure/bicep-registry-modules#393]
### Module path

CONTRIBUTING.md

### Bicep version

N/A

### Describe the bug

The contribution guidance says that version.json support major.minor.patch, but the regex doesn't allow a patch number. If I'm reading the regex, you allow `<major>.<minor>-<any-string><number>-<any-string>`.  I'm assuming this is to support `-preview` or similar strings, which is great, but why not support a patch number like the docs say? Can you bring back patch please?

### To reproduce

Create a version.json with a `1.0.1` version number.

### Code snippet

```Bicep
{
  "$schema": "https://aka.ms/bicep-registry-module-version-file-schema#",
  "version": "1.0.1",
  "pathFilters": [
    "./main.json",
    "./metadata.json"
  ]
}
```

### Relevant log output

```Shell
The file "/home/runner/work/bicep-registry-modules/bicep-registry-modules/modules/cost/resourcegroup-scheduled-action/version.json" is invalid:
  #/version: Value does not match the pattern of "^v?(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(-(?:[\\da-z\\-]+)(?:\\.(?:[\\da-z\\-]+))*)?$"

Error: Process completed with exit code 1.

See https://github.com/Azure/bicep-registry-modules/actions/runs/5123310355/jobs/9213629301?pr=300
```

**Comment by shenglol** [comment]
Hi @flanakin - this is by design. I wanted to provide some insights regarding the reasoning behind the design decisions related to PATCH versioning in the context of module development. Our intention is to ensure consistency and avoid the situation where module authors set PATCH versions freely. Additionally, we aim to prevent registry modules from having version suffixes such as `-beta` or `-preview`.

According to the guidelines mentioned in the contribution guide [here](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#bumping-patch-version), the PATCH version is automatically assigned once a module is merged into the main branch. This assignment is based on the "height" of the git commits since the last modification of the main.json or metadata.json file within the module on the main branch. Because we only allow squash merging, the PATCH version will always be increased by 1. This is essentailly done with a tool called [Nerdbank.GitVersioning](https://github.com/dotnet/Nerdbank.GitVersioning), which is also used for version management in the main Bicep repository.
